### PR TITLE
EBL-420:fix RequestBlocker to block and not only to render the message

### PR DESCRIPTION
**Pull request by maorshap**

### CODE REVIEW

**Review by matvey-mtn — APPROVED** [review]

File: apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java [hunk 1]
```diff
@@ -116,6 +116,8 @@ public void addDeployment(String environmentIdsCsv, String serviceIdsCsv, int de
             deploymentHandler.checkDeploymentShouldBeBlockedByRequestBlocker(serviceIds, environmentIds.size());
         } catch (ApolloDeploymentException e) {
             responseObject.addUnsuccessful(e);
+            assignJsonResponseToReq(req, HttpStatus.CREATED, responseObject);
```
Comment (mikeskali) [inline apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java:119]: why CREATED? don't you want to return an error?

File: apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentGroupsController.java [hunk 2]
```diff
@@ -43,6 +43,8 @@ public void addDeployment(int environmentId, int serviceId, int deployableVersio
             deploymentHandler.checkDeploymentShouldBeBlockedByRequestBlocker(new ArrayList<Integer>() {{ add(serviceId); }}, 1);
         } catch (ApolloDeploymentException e) {
             responseObject.addUnsuccessful(e);
+            assignJsonResponseToReq(req, HttpStatus.CREATED, responseObject);
```
Comment (mikeskali) [inline apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentGroupsController.java:46]: same

File: apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java (hunk 1 above)
Comment (maorshap) [reply apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java:119]: @mikeskali Changed to Internal error

Comment (mikeskali) [reply apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java:119]: Not internal error as well. Nothing is wrong internally. Bad request.

Comment (maorshap) [reply apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java:119]: @mikeskali Yeah sorry I mean bad request :).

**Review by mikeskali — APPROVED** [review]

**Pull request merged**